examples/keras_recipes/creating_tfrecords.py

```diff
@@ -0,0 +1,193 @@
+"""
+Title: Creating TFRecords
+Author: [Dimitre Oliveira](https://www.linkedin.com/in/dimitre-oliveira-7a1a0113a/)
+Date created: 2021/02/22
+Last modified: 2021/02/22
+Description: Converting data to TFRecords files.
+"""
+
+"""
+## Reference
+
+- [TFRecord and tf.train.Example](https://www.tensorflow.org/tutorials/load_data/tfrecord)
+
+## Introduction
+
+The TFRecord format is a simple format for storing a sequence of binary records.
+
+Converting your data into TFRecords has many advantages, especially if later, you load it
+using the [tf.data](https://www.tensorflow.org/guide/data) API. The data will take less
+space from disk, it can also be read and processed faster.
+
+Here we will learn how to convert data of different types (image, text, and numeric) into
+TFRecords.
+
+## Dependencies
+"""
+
+import os
+import json
+import pprint
+import tensorflow as tf
+import matplotlib.pyplot as plt
+
+"""
+## Downloading the COCO2017 dataset
+
+We will be using the [COCO2017](https://cocodataset.org/) dataset because it has many
+different types of features and will serve as a good example of how to encode different
+features as TFRecords.
+"""
+
+root_dir = "datasets"
+tfrecords_dir = "tfrecords"
+images_dir = os.path.join(root_dir, "val2017")
+annotations_dir = os.path.join(root_dir, "annotations")
+annotation_file = os.path.join(annotations_dir, "instances_val2017.json")
+images_url = "http://images.cocodataset.org/zips/val2017.zip"
+annotations_url = (
+    "http://images.cocodataset.org/annotations/annotations_trainval2017.zip"
+)
+
+# Download image files
+if not os.path.exists(images_dir):
+    image_zip = tf.keras.utils.get_file(
+        "images.zip", cache_dir=os.path.abspath("."), origin=images_url, extract=True,
+    )
+    os.remove(image_zip)
+
+# Download caption annotation files
+if not os.path.exists(annotations_dir):
+    annotation_zip = tf.keras.utils.get_file(
+        "captions.zip",
+        cache_dir=os.path.abspath("."),
+        origin=annotations_url,
+        extract=True,
+    )
+    os.remove(annotation_zip)
+
+print("Dataset is downloaded and extracted successfully.")
+
+with open(annotation_file, "r") as f:
+    annotations = json.load(f)["annotations"]
+
+print(f"Number of images: {len(annotations)}")
+
+"""
+## Looking at the annotations of a single data sample
+"""
+
+pprint.pprint(annotations[60])
+
+"""
+## Parameters
```

Comment:
We should add some explanatory text

Reply:
Text added with more details.

Reply:
Moved this section to be right after the dataset features description, this way they are complementary.